src/Features/CSharp/Portable/Completion/KeywordRecommenders/RefKeywordRecommender.cs

```diff
@@ -67,6 +67,27 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
                 IsValidNewByRefContext(syntaxTree, position, context, cancellationToken);
         }
 
+        private static bool IsRefParameterModifierContext(int position, CSharpSyntaxContext context)
+        {
+            if (context.SyntaxTree.IsParameterModifierContext(
+                    position, context.LeftToken, includeOperators: false, out var parameterIndex, out var previousModifier))
+            {
+                if (previousModifier == SyntaxKind.None)
+                {
+                    return true;
+                }
+
+                if (previousModifier == SyntaxKind.ThisKeyword &&
+                    parameterIndex == 0 &&
+                    context.SyntaxTree.IsPossibleExtensionMethodContext(context.LeftToken))
```

Comment:
as above, it seems reasonable to just have this be: `return prev == SyntaxKind.None || prev == SyntaxKind.This`.